### ISSUE TITLE
Cannot use OpenRouter's free models

### ISSUE THREAD

**Issue by shiyi-jiaqiu** [RooVetGit/Roo-Code#245]
### Which API Provider are you using?

OpenRouter

### Which Model are you using?

google/gemini-2.0-flash-exp:free

### What happened?

![image](https://github.com/user-attachments/assets/c7616ffd-1bd8-402b-a657-77c6c7fad16c)
Display this model as invalid, please allow custom model names in the openrouter api provider

### Steps to reproduce

1.
2.
3.

**Comment by samhvw8** [comment]
<img width="538" alt="image" src="https://github.com/user-attachments/assets/17e4ee59-4a5a-4d1d-b9b9-265a8c3070e5" />
 i don't have your error, what your roo-cline version, how to reproduce your bug ?

**Comment by shiyi-jiaqiu** [comment]
v2.2.34; I found that when using cline or Roo-cline on Windows, a selection box can appear for model selection.
![image](https://github.com/user-attachments/assets/95b8c348-8ad5-42ab-858b-32fc94971b62)
But in WSL2 (Ubuntu 22.04), both cline and Roo-cline encounter the above issues, with no selection box, only claude as an option. 
![image](https://github.com/user-attachments/assets/e7c46253-5082-4e93-9fbb-23d5b237a895)

**Comment by tonygao2k** [comment]
same issue on my side (Mac), cannot select other models but claude-3.5-sonnet when using OpenRouter.

**Comment by mrubens** [comment]
Hmm, it seems like it might be an issue with this API call failing. Any chance you have a firewall or something that would be blocking this?
https://github.com/RooVetGit/Roo-Cline/blob/c30e9c6ed3eb733a566c84a4ae3f4aa4a4e67f92/src/core/webview/ClineProvider.ts#L1047

**Comment by samhvw8** [comment]
in some case, ipv6 setup not correctly in wsl, you can try to disable ipv6 to test it again

**Comment by tonygao2k** [comment]
> Hmm, it seems like it might be an issue with this API call failing. Any chance you have a firewall or something that would be blocking this?
> 
> https://github.com/RooVetGit/Roo-Cline/blob/c30e9c6ed3eb733a566c84a4ae3f4aa4a4e67f92/src/core/webview/ClineProvider.ts#L1047

Yes, I have VPN on my side, but i tried curl "https://openrouter.ai/api/v1/models" on terminal, it can get correct response.

**Comment by tonygao2k** [comment]
![error](https://github.com/user-attachments/assets/235e44f9-5a39-40ab-822d-874c347fa61b)
DEBUG info, seems network issue but why "curl 'https://openrouter.ai/api/v1/models'" can get correct response on vscode terminal.

**Comment by mrubens** [comment]
@tonygao2k anything interesting in the network tab of the developer tools?

**Comment by tonygao2k** [comment]
> @tonygao2k anything interesting in the network tab of the developer tools?
![error2](https://github.com/user-attachments/assets/0c6c2c3a-6296-47e7-8104-dad50e607419)

it's blank

**Comment by mrubens** [comment]
Which editor are you using?

**Comment by tonygao2k** [comment]
![vscode](https://github.com/user-attachments/assets/d6a883da-0a45-46c1-919f-951096c89341)
this version

**Comment by tonygao2k** [comment]
![ddd](https://github.com/user-attachments/assets/28c20a6a-27c4-4a1d-864d-55f311c0af4b)

I have just to delete the model text field but still failed to see other models but only claude-sonnet-3.5

**Comment by samhvw8** [comment]
@tonygao2k  can you give us screenshoot of reponse 
```
curl -6 https://openrouter.ai/api/v1/models

curl -4 https://openrouter.ai/api/v1/models
```

**Comment by tonygao2k** [comment]
> @tonygao2k can you give us screenshoot of reponse
> 
> ```
> curl -6 https://openrouter.ai/api/v1/models
> 
> curl -4 https://openrouter.ai/api/v1/models
> ```

[curl-4.json](https://github.com/user-attachments/files/18357999/curl-4.json)
[curl-6.json](https://github.com/user-attachments/files/18358001/curl-6.json)

Please find attachment, these two commands returned  invalid json

**Comment by tonygao2k** [comment]
> @tonygao2k can you give us screenshoot of reponse
> 
> ```
> curl -6 https://openrouter.ai/api/v1/models
> 
> curl -4 https://openrouter.ai/api/v1/models
> ```

![image](https://github.com/user-attachments/assets/851415ba-fd1d-4d2c-937b-7a9a56df917c)

[curl-4.json](https://github.com/user-attachments/files/18358025/curl-4.json)
[curl-6.json](https://github.com/user-attachments/files/18358026/curl-6.json)

Please find attachment, these two commands can return correct result on vscode terminal. 

As i said, i am using VPN on my side otherwise "https://openrouter.ai/" is unreachable. Can you please check if Roo-Cline applied "http_proxy" correctly from environment variable or settings.json?

**Comment by tonygao2k** [comment]
![image](https://github.com/user-attachments/assets/74d8cac0-7a61-40ae-bb46-f68cf367f114)

@samhvw8 @mrubens , attached is workaround to solve this issue on my side.  adding proxy support for "refreshOpenRouterModels()" in ClineProvider.ts
[Uploading workaround.txt…]()

**Comment by XIONGPEILIN** [comment]
My proxy cannot respond to IPv6, which is probably the reason for encountering the same issue. (cu12) xiong-p@gp42:~/gazo$ curl -6 https://openrouter.ai/api/v1/models
curl: (5) Couldn't resolve proxy 'proxy.uec.ac.jp'
